src/Symfony/Component/DependencyInjection/ParameterBag/ContainerBag.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\ParameterBag;
+
+use Symfony\Component\DependencyInjection\Container;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class ContainerBag extends FrozenParameterBag implements ContainerBagInterface
```

Comment:
Could probably be made final, I see no reason to extend this rather than implementing the interface.

Reply:
We don't make classes final "by default", that's not the framework's policy.

Reply:
Could that change for new classes?
Like adding scalar type is?